src/containers/checkout/delivery/delivery-type/delivery-type.js

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { FormControl, InputLabel, Select, MenuItem } from '@material-ui/core';
+import { useStyles } from '../../checkout.styles';
+import {
+  CHECKOUT_DELIVERY_TYPES,
+  CHECKOUT_DROP_LIST
+} from '../../../../translations/checkout.translations';
+import DeliveryInfo from './delivery-info/delivery-info';
+import { SelfPickupTop, SelfPickupBottom } from './mail-services/self-pickup';
+import { NovaPoshtaTop, NovaPoshtaBottom } from './mail-services/nova-poshta';
+import { UkrposhtaTop, UkrPoshtaBottom } from './mail-services/ukrposhta';
+import { CurrierBottom } from './mail-services/currier/currier-bottom';
+import { getNovaPoshtaCities } from '../../../../redux/checkout/checkout.actions';
+
+const DeliveryType = ({ deliveryType, setDeliveryType }) => {
+  const style = useStyles();
+  const { language, contacts } = useSelector(({ Language, Contacts }) => ({
+    language: Language.language,
+    contacts: Contacts.contacts
+  }));
+
+  const [city, setCity] = useState('');
+  const dispatch = useDispatch();
+
+  const { cities } = useSelector(({ Checkout }) => ({
+    cities: Checkout.cities
+  }));
+
+  const citiesForNovaPoshta = cities.map((citi) => citi.description);
```

Comment:
I think firstly we should check if `cities` exist and before we manipulate with it

Reply:
done